app/grandchallenge/cases/image_builders/dicom.py

```diff
@@ -167,49 +168,56 @@ def _process_dicom_file(dicom_ds):  # noqa: C901
     content_times = []
     exposures = []
 
+    origin = None
+    origin_diff = (0.0, 0.0, 0.0)
+    n_origins = 0
+    for partial in dicom_ds.headers:
+        ds = partial["data"]
+        if "ImagePositionPatient" in ds:
+            file_origin = tuple(float(i) for i in ds.ImagePositionPatient)
+            if origin:
+                diff = tuple(np.subtract(file_origin, origin))
+                origin_diff = tuple(np.add(origin_diff, diff))
+            origin = file_origin
+            n_origins += 1
+    avg_origin_diff = tuple(x / n_origins for x in origin_diff)
```

Comment:
since you are using numpy anyway, you can write it straightup in the numpy way - prroabaly more efficient and everything:

```
origin = None
origin_diff = np.array((0, 0, 0), dtype=float)
n_origins = 0
for partial in dicom_ds.headers:
    ds = partial["data"]
    if "ImagePositionPatient" in ds:
        file_origin = np.array(ds.ImagePositionPatient, dytpe=float)
        if origin:
            diff = file_origin - origin
            origin_diff = origin_diff + diff
        origin = file_origin
        n_origin += 1
avg_origin_diff = tuple(origin_diff / n_origins)
```

Just a little tip for the future. more readable and less conversions between datatypes this way...

Reply:
Thank you. I definitely need to learn to make better use of numpy's capabilities.